cmake/algebra-plugins-compiler-options-cuda.cmake

```diff
@@ -0,0 +1,27 @@
+# Algebra plugins library, part of the ACTS project (R&D line)
+#
+# (c) 2021-2022 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+# FindCUDAToolkit needs at least CMake 3.17.
+cmake_minimum_required( VERSION 3.17 )
+
+# Include the helper function(s).
+include( algebra-plugins-functions )
+
+# Figure out the properties of CUDA being used.
+find_package( CUDAToolkit REQUIRED )
+
+# Set the architecture to build code for.
+set( CMAKE_CUDA_ARCHITECTURES "52" CACHE STRING
```

Comment:
This was a big contention point with @stephenswat in [vecmem](https://github.com/acts-project/vecmem) a while ago. I just can't even find the PR about it anymore, that's how long ago it was...

This is meant to work around a warning from modern CMake versions. Note that vecmem has been setting this cache variable to this value since a very long time. So unless you were overriding it with `-DCMAKE_CUDA_ARCHITECTURES=...` in your builds, you've been making all of your builds for this one architecture all along...

We will absolutely need to explore how choosing different "build architectures" on different "hardware architectures" would affect the performance of our code.

But for now let's just put this in here. It will literally not make any difference for the builds at the moment, as vecmem is anyway polluting all the other projects with this setting already.

Reply:
OK. let's get back to this in the future. Actually, I have been aware that specifying architecture number actually doesn't affect the performance a lot